apps/discord-bot/src/listeners/events/auto-thread.test.ts

```diff
@@ -0,0 +1,96 @@
+import { Client, Events, TextChannel } from "discord.js";
+import {
+  mockTextChannel,
+  mockMessage,
+  mockGuildMember,
+  emitEvent,
+} from "@answeroverflow/discordjs-mock";
+import { setupAnswerOverflowBot } from "~discord-bot/test/sapphire-mock";
+import { randomSnowflake } from "@answeroverflow/discordjs-utils";
+import { mockPublicThread } from "@answeroverflow/discordjs-mock";
+
+let client: Client;
+let textChannel: TextChannel;
+
+beforeEach(async () => {
+  client = await setupAnswerOverflowBot();
+  textChannel = mockTextChannel(client);
+});
+
+describe("Auto thread", () => {
+  it("should not create a thread on a channel thread channel", async () => {
+    const channel = mockPublicThread({ client });
+    const author = mockGuildMember({ client });
+    const message = mockMessage({
+      client,
+      channel,
+      author: author.user,
+    });
+
+    await emitEvent(client, Events.MessageCreate, message);
+
+    expect(message.startThread).not.toHaveBeenCalled();
+  });
+  it("should not create a thread if the author is a bot", async () => {
```

Comment:
For this test, we should make it so all other factors succeed except for it being a bot

A way to test this is to remove the .not from toHaveBeenCalled, update the test to it passes, and then add the .not back and change the caller back to a bot so it fails

Reply:
So we test everything normally (like with the user being a human, and the message is in a text channel with the correct flags)? Then add another expect afterwards in the same `it` block with the user being a bot?

Reply:
Yup! Is there any way we can keep this dry? We we'll have the same code across all the tests?

Reply:
Feel free to refactor it if you want to but personally when I'm writing tests I don't really care to much about keeping it DRY

Most of the tests are testing that specific scenario, so I prefer to just keep it explicit what is being tested and then also let copilot generate the tests for me

They don't need to look pretty all they have to do is sit there and ensure that this stuff doesnt break as updates happen